crates/consensus/src/block.rs

```diff
@@ -0,0 +1,38 @@
+//! Genesic Block Type
+
+use crate::{Header, Requests};
+use alloc::vec::Vec;
+use alloy_eips::{
+    eip2718::{Decodable2718, Encodable2718},
+    eip4895::Withdrawal,
+};
+use alloy_rlp::{Decodable, Encodable, RlpDecodable, RlpEncodable};
+
+/// Ethereum full block.
+///
+/// Withdrawals can be optionally included at the end of the RLP encoded message.
+///
+/// Taken from [reth-primitives](https://github.com/paradigmxyz/reth)
+#[derive(Debug, Clone, PartialEq, Eq, Default, RlpEncodable, RlpDecodable)]
+#[rlp(trailing)]
+pub struct Block<T: Encodable + Decodable + Encodable2718 + Decodable2718> {
```

Comment:
this should only use `Encodable + Decodable`

and it is expected that this is equivalent to 

https://github.com/alloy-rs/alloy/blob/21bc07ebbcb327486fc260cb1a8fb9d23f4bae53/crates/eips/src/eip2718.rs#L219-L219

see:

https://github.com/alloy-rs/alloy/blob/21bc07ebbcb327486fc260cb1a8fb9d23f4bae53/crates/consensus/src/transaction/envelope.rs#L329-L332

because this type is the p2p type